### DISCUSSION TITLE
Only show screen when value above 0

### DISCUSSION THREAD

**Discussion by mciaio** [pixelit-project/PixelIt#231] [General]
I have several screens working and all based on a value. Let's take solar power as an example. I want it to update and show the solar watts from panels only above 0 watts. When it is 0 watts, I want it to not show in the rotation. I mostly have it working, but instead of going away at 0, it reads the last value forever which is normally 1 watt. I noticed this is possible from the 'days til christmas" screen but can't duplicate it. I am pulling the values from MQTT, is that the problem?

I am using a function node with this code:

`var percent = msg.payload

msg.payload = percent;

if (percent === 0) {
msg.show = false
}
else if (percent > 0) {
return msg;
}`

**Comment by hamster65** [comment]
Your function only returns something when percent is > 0. It does not do anything when percent is 0, which is why PixelIt shows the last value > 0.

I would add some debug statements using node.warn, to figure out the code flow and the contents of msg.payload and percent. 
Described here: https://nodered.org/docs/user-guide/writing-functions

**Comment by mciaio** [comment]
Thanks, I appreciate the input.  Do you have any suggestion on how to make
it work the way I would like it to?  I want the screen out of the rotation
when the value is 0.

**Comment by hamster65** [comment]
I assume your value is in msg.payload at the beginning?
I would try it like this:

var percent = msg.payload;

if (percent > 0) {
  msg.show = true;
  node.status({fill:"green",shape:"dot",text:"percent > 0, show"});
} else {
 msg.show = false;
  node.status({fill:"red",shape:"dot",text:"percent <= 0, hide"});
}
return msg;

**Comment by mciaio** [comment]
Awesome!  Thanks!

**Comment by davosje** [comment]
Hi! i'm trying to get my solar information from HA to my clock. It works sometimes but only get undefined as information.

This is what I have:

![image](https://user-images.githubusercontent.com/10655133/212346014-b98709f8-b987-4d76-8eb8-3534ceb9d9fb.png)

![image](https://user-images.githubusercontent.com/10655133/212334488-a49ee24e-4c52-4411-acf2-771925456c57.png)

Solar total HA is just a number represtenting watts.

with function node:

```
var watt = msg.payload

if (watt > 0) {
    msg.show = true;
    node.status({ fill: "green", shape: "dot", text: "watt > 0, show" });
} else {
    msg.show = false;
    node.status({ fill: "red", shape: "dot", text: "watt <= 0, hide" });
}
return msg;
```

I'm doing it wrong but have no idea, any suggestions?

**Comment by mciaio** [comment]
I don't know if it matters, but I have screen init prior to text.  Swap that, then put a debug module after text and see what it's spitting out.

Your function code looks like it should work to me.

**Comment by davosje** [comment]
Thanks! Swapped it and used the debug node. Its spitting out a value. What should be the 'Text' in the last text node? Clock shows icon and as value undefined.

EDIT: got it, should be {{payload}}

![image](https://user-images.githubusercontent.com/10655133/212473148-8ea9686f-a3fa-45b5-a00a-66c8619442dd.png)

**Comment by mciaio** [comment]
{{msg.payload}}

**Comment by aletdf** [comment]
Hi! I was reading this thread and I am interested to see the results in your screen. Can you share a picture or short video pleas?
Thanks!
